### PR TITLE
feat: rewrite VueLoaderPlugin to support webpack5

### PR DESCRIPTION
**Pull request by 18566246732**

### CODE REVIEW

File: lib/plugin.js [hunk 1]
```diff
@@ -104,56 +128,60 @@ class VueLoaderPlugin {
...
-    // #1201 we need to skip the `include` check when locating the vue rule
-    const clone = Object.assign({}, rule)
-    delete clone.include
-    const normalized = RuleSet.normalizeRule(clone, {}, '')
-    return (
-      !rule.enforce &&
-      normalized.resource &&
-      normalized.resource(fakeFile)
-    )
+function cloneRule (rawRule, ruleSetCompiler, refs) {
```
Comment (sodatea) [inline lib/plugin.js]: There's only one instance of `ruleSetCompiler` in this module so no need to pass it around.

File: lib/plugin.js [hunk 2]
```diff
@@ -104,56 +128,60 @@ class VueLoaderPlugin {
...
+  const rules = ruleSetCompiler.compileRules("ruleSet", [{
+    rules: [rawRule]
+  }], refs)
+  let currentResource
+
+  const conditions = rules[0].rules
+    .map(rule => rule.conditions)
+    .flat();
```
Comment (sodatea) [inline lib/plugin.js]: `.flat` is only supported in Node.js v11+, so don't use it.

File: lib/plugin.js [hunk 3]
```diff
@@ -104,56 +128,60 @@ class VueLoaderPlugin {
...
+
+  // do not process rule with enforce
+  if (!rawRule.enforce) {
+    const ruleUse = rules[0].rules
+      .map(rule => rule.effects
+        .filter(effect => effect.type === 'use')
+        .map(effect => effect.value)
+      )
+      .flat()
```
Comment (sodatea) [inline lib/plugin.js]: Don't use `.flat`, for compatibility concerns.

**Comment by sodatea** [comment]
😰I'm so sorry I seemed to have messed things up by accidentally force pushing to your fork.

Could you open a new pull request, please?

**Comment by 18566246732** [comment]
okay, no problem, I'll make a new pr

**Pull request closed without merging**